src/main/java/io/debezium/connector/spanner/db/metadata/SchemaRegistry.java

```diff
@@ -89,38 +94,70 @@ private boolean validate(TableId tableId, List<Column> rowType) {
+    @VisibleForTesting
+    void setSpannerSchema(SpannerSchema spannerSchema) {
+        this.spannerSchema = spannerSchema;
+    }
 
-        if (this.changeStream == null) {
-            throw new IllegalStateException("Change stream doesn't exist at timestamp: "
-                    + streamName + ", " + timestamp);
-        }
+    public void updateSchemaFromStaleTimestamp(TableId tableId, Timestamp timestamp, List<Column> rowType) {
+        LOGGER.info("Schema is outdated. Try to update schema registry outside of retention period...");
+        TableSchema watchedTable = this.getWatchedTable(tableId);
+        SpannerSchema.SpannerSchemaBuilder builder = SpannerSchema.builder();
+        for (Column column : rowType) {
+            if (column.isPrimaryKey()) {
+                builder.addPrimaryColumn(watchedTable.getName(), column.getName());
+            }
+            builder.addColumn(watchedTable.getName(), column.getName(), column.getType().getType().toString(), column.getOrdinalPosition(),
+                    column.isNullable(), this.schemaDao.getDatabaseClient().getDialect());
 
-        SpannerSchema newSchema;
-        if (this.changeStream.isWatchedAllTables()) {
-            newSchema = schemaDao.getSchema(timestamp);
         }
-        else {
-            newSchema = schemaDao.getSchema(timestamp, this.changeStream.getTables());
+        this.spannerSchema = SchemaMerger.merge(this.spannerSchema, builder.build());
+        LOGGER.info("Schema registry has been updated to stale timestamp {}", timestamp);
+
+    }
+
+    @VisibleForTesting
+    void forceUpdateSchema(@Nullable TableId tableId, Timestamp updatedTimestamp, @Nullable List<Column> rowTypes) {
+        try {
+            this.timestamp = updatedTimestamp;
+            this.changeStream = schemaDao.getStream(timestamp, streamName);
+
+            if (this.changeStream == null) {
+                throw new IllegalStateException("Change stream doesn't exist at timestamp: "
+                        + streamName + ", " + timestamp);
+            }
+
+            SpannerSchema newSchema;
+            if (this.changeStream.isWatchedAllTables()) {
+                newSchema = schemaDao.getSchema(timestamp);
+            }
+            else {
+                newSchema = schemaDao.getSchema(timestamp, this.changeStream.getTables());
+            }
+            if (this.spannerSchema == null) {
+                this.spannerSchema = newSchema;
+                return;
+            }
+            this.spannerSchema = SchemaMerger.merge(this.spannerSchema, newSchema);
         }
-        if (this.spannerSchema == null) {
-            this.spannerSchema = newSchema;
-            return;
+        catch (SpannerException e) {
+            // Error code 9 is for FAILED_PRECONDITION when tries to read out of version retention period.
+            if (e.getMessage().contains("has exceeded the maximum timestamp staleness") && e.getCode() == 9) {
```

Comment:
Can you match the error code based on the actual string? 

Example: https://github.com/debezium/debezium-connector-spanner/blob/bde809e0163a5e1c4ca3816a74e726f5ce8fc3f9/src/main/java/io/debezium/connector/spanner/db/stream/SpannerChangeStream.java#L220